server/routes/api/chart.py

```diff
@@ -213,48 +210,74 @@ def choropleth_data(dcid):
+    configs = get_choropleth_configs()
+    stat_vars, denoms = get_stat_vars(configs)
+    display_dcid, display_level = get_choropleth_display_level(dcid)
+    geos = dc_service.get_places_in([display_dcid],
+                                    display_level).get(display_dcid, [])
+    if not stat_vars or not geos:
         return Response(json.dumps({}), 200, mimetype='application/json')
     # Get data for all the stat vars for every place we will need and process the data
-    all_sv_data = dc_service.get_stats_all(geos, list(all_stat_vars))
-    if not 'placeData' in all_sv_data:
-        return Response(json.dumps({}), 200, mimetype='application/json')
-    processed_data = process_choropleth_data(all_sv_data['placeData'])
-
+    sv_data = dc_service.get_stat_set_within_place(display_dcid, display_level,
+                                                   list(stat_vars),
+                                                   "").get('data', {})
+    denoms_data = {}
+    for denom in denoms:
+        denoms_data[denom] = dc_service.get_stats(geos, denom)
     result = {}
-    for cc in choropleth_configs:
+    for cc in configs:
         # we should only be making choropleths for configs with a single stat var
         sv = cc['statsVars'][0]
-        cc_data, statvar_denom = landing_page_api.get_snapshot_across_places(
-            cc, processed_data, geos)
-        data_values = cc_data.get('data', [])
-        data_dict = dict()
+        cc_sv_data_values = sv_data.get(sv, {}).get('stat', {})
+        cc_sv_data_metadata = sv_data.get(sv, {}).get('metadata', {})
+        denom = landing_page_api.get_denom(cc, True)
+        cc_denom_data = denoms_data.get(denom, {})
         scaling = cc.get('scaling', 1)
         if 'relatedChart' in cc:
             scaling = cc['relatedChart'].get('scaling', scaling)
-        for value in data_values:
-            if 'dcid' not in value or 'data' not in value:
+        sources = set()
+        dates = set()
+        data_dict = dict()
+        for place_dcid in cc_sv_data_values:
+            dcid_sv_data = cc_sv_data_values.get(place_dcid)
+            val = dcid_sv_data.get("value", None)
+            if not val:
                 continue
-            val = value['data'].get(sv, None)
-            if val:
-                val = val * scaling
-            data_dict[value['dcid']] = val
+            date = dcid_sv_data.get("date", "")
+            dates.add(date)
+            import_name = dcid_sv_data.get("metadata", {}).get("importName", "")
+            source = cc_sv_data_metadata.get(import_name,
+                                             {}).get("provenanceUrl", "")
+            sources.add(source)
+            if denom:
+                if not cc_denom_data.get(place_dcid, {}):
+                    continue
+                denom_val = get_denom_val(
+                    date, cc_denom_data[place_dcid].get('data', {}))
+                val = val / denom_val
```

Comment:
check for value of denom_val before dividing?

Reply:
done